### PR TITLE
Fix test return type for psr/container v2

### PR DESCRIPTION
**Pull request by internalsystemerror**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

This add return types to the tests where `ContainerInterface` is used allowing https://github.com/mezzio/mezzio/pull/121 to be merged.

### CODE REVIEW

File: src/Container/ApplicationConfigInjectionDelegator.php [hunk 1]
```diff
@@ -117,6 +117,7 @@ public static function injectPipelineFromConfig(Application $application, array
         }
 
         // Create a priority queue from the specifications
+        /** @var SplPriorityQueue $queue */
```
Comment (Ocramius) [inline src/Container/ApplicationConfigInjectionDelegator.php]: createPriorityQueueReducer return type should be documented better here

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Container/ApplicationConfigInjectionDelegator.php (hunk 1 above)
Comment (internalsystemerror) [reply src/Container/ApplicationConfigInjectionDelegator.php]: Yeah, it means using templates/generic types and I could get carried away at that

Comment (Ocramius) [reply src/Container/ApplicationConfigInjectionDelegator.php]: Nah, just declare that it returns a `callable(...): SplPriorityQueue` 😜

This `@var` should go away

Comment (internalsystemerror) [reply src/Container/ApplicationConfigInjectionDelegator.php]: so it did!

**Review by Ocramius — APPROVED** [review]
Noice! 🚢

**Pull request merged**